src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -18,23 +18,60 @@
 import com.google.maps.model.PlacesSearchResponse;
 import com.google.maps.model.PlacesSearchResult;
 import java.util.logging.Logger;
+import java.util.Iterator;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** BusinessesService object representing all businesses 
 * components of the webapp.
  **/
 public class BusinessesService {
   
-  private List<Listing> businesses;
+  private List<Listing> allBusinesses;
   private final String KEY = "REDACTED";
   private final static Logger LOGGER = 
         Logger.getLogger(BusinessesService.class.getName());
   private final int ALLOWED_SEARCH_REQUESTS = 3;
 
   /** Create a new Businesses instance
-  * @param businesses businesses from SmallCityService
+  * @param allBusinesses businesses from SmallCityService
   **/
-  public BusinessesService(List<Listing> businesses) {
-    this.businesses = businesses;
+  
+  public BusinessesService(List<Listing> allBusinesses) {
+    this.allBusinesses = allBusinesses;
+  }
+
+  public PreparedQuery getBigBusinessFromDatabase(){
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("BigBusinesses");
+    PreparedQuery queryOfDatabase = datastore.prepare(query);
+    return queryOfDatabase;
+  }
+
+  public List<Listing> removeBigBusinessesFromResults(PreparedQuery queryOfDatabase){
+    Iterator<Listing> smallBusinesses =  allBusinesses.iterator();
+    Entity entity;
+    String businessName;
+    while (smallBusinesses.hasNext()) {
+      Listing currentBusiness = smallBusinesses.next();
+      Iterator<Entity> bigBusinessEntities = queryOfDatabase.asIterator();
+      while(bigBusinessEntities.hasNext()) {
+        businessName = 
+              (String) bigBusinessEntities.next().getProperty("Business");
+        if(businessName.equals(currentBusiness.getName())) {
+          smallBusinesses.remove();
+        }
+      }
+    }
+    return allBusinesses;
+  }
+
+  public void setAllBusinesses(List<Listing> allBusinesses) {
+   this.allBusinesses = allBusinesses;
   }
```

Comment:
This functionality is already taken care of in the constructor.

Reply:
fixed